packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -0,0 +1,286 @@
+  public readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+  // this works for all 3 cluster engines that are currently supported
+  public readonly defaultInstanceType = ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.MEDIUM);
+
+  private readonly needsS3RolesInParameters: boolean;
+  private readonly parameterGroupFamilies?: ParameterGroupFamilyMapping[];
+  private readonly defaultPort?: number;
+
+  constructor(
+    engineName: string,
+    needsS3RolesInParameters: boolean,
+    singleUserRotationApplication: secretsmanager.SecretRotationApplication,
+    multiUserRotationApplication: secretsmanager.SecretRotationApplication,
+    parameterGroupFamilies?: ParameterGroupFamilyMapping[],
+    defaultPort?: number,
+    engineVersion?: string) {
+
+    this.engineType = engineName;
+    this.needsS3RolesInParameters = needsS3RolesInParameters;
+    this.singleUserRotationApplication = singleUserRotationApplication;
+    this.multiUserRotationApplication = multiUserRotationApplication;
+    this.parameterGroupFamilies = parameterGroupFamilies;
+    this.defaultPort = defaultPort;
+    this.engineVersion = engineVersion;
+    this.parameterGroupFamily = this.establishParameterGroupFamily();
+  }
+
+  public withVersion(engineVersion: string): IClusterEngine {
+    const self = this;
+    // put it in a local variable, as it's protected
+    const defaultParameterGroup = this.defaultParameterGroup;
+
+    class CopyWithVersion extends AbstractClusterEngine {
+      protected defaultParameterGroup(scope: core.Construct) {
+        // we need to use call, so that 'this' inside defaultParameterGroup() is resolved correctly
+        return defaultParameterGroup.call(self, scope);
+      }
+    }
+
+    return new CopyWithVersion(
+      this.engineType,
+      this.needsS3RolesInParameters,
+      this.singleUserRotationApplication,
+      this.multiUserRotationApplication,
+      this.parameterGroupFamilies,
+      this.defaultPort,
+      engineVersion,
+    );
+  }
+
+  public bindToCluster(scope: core.Construct, options: ClusterEngineBindOptions): ClusterEngineBindConfig {
+    let parameterGroup: IParameterGroup | undefined;
+    if (options.parameterGroup) {
+      parameterGroup = options.parameterGroup;
+    } else if (this.needsS3RolesInParameters && (options.s3ImportRole || options.s3ExportRole)) {
+      // in this case, we need to create a new ParameterGroup to store the S3 Role parameters in
+      // (imported ParameterGroups, like the RDS default ones, don't allow adding parameters to them)
+      parameterGroup = new ClusterParameterGroup(scope, 'ClusterParameterGroup', {
+        family: this.parameterGroupFamily,
+      });
+    } else {
+      parameterGroup = this.defaultParameterGroup(scope);
+    }
+
+    if (this.needsS3RolesInParameters) {
+      if (options.s3ImportRole) {
+        parameterGroup?.addParameter('aurora_load_from_s3_role', options.s3ImportRole.roleArn);
```

Comment:
Can this only ever be true for Aurora? If so, shouldn't this live in the Aurora subclass?

Reply:
No. It's also true for Aurora MySQL.